### ISSUE TITLE
Bug on sign in with Chrome Custom Tabs

### ISSUE THREAD

**Issue by maor121** [okta/okta-oidc-android#110]
Hi,

I build my android app using Android SDK v28, and android okta library 'com.okta.android:oidc-android:1.0.4'.
I try to sign in with Chrome Custom Tabs using the tutorial:
https://developer.okta.com/docs/guides/sign-into-mobile-app/android/sign-in-page/

And I get AuthorizationException : "No compatible browser found'. (even though I have chrome installed)

Android device info:
Xiaomi Redmi Note 6 pro
Android version: 9 PKQ1.180904.001
Installed browsers: default browser + Chrome

I debugged okta source code, and the solution ended being simple. But it requires source code change:

The class "OktaAuthenticationActivity" has a lines:
`Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.example.com"));
List<ResolveInfo> resolveInfoList = pm.queryIntentActivities(browserIntent, 0);`

The resolveInfoList  should return 2 results, for the 2 browsers I have. But they return only the default browser, which does not support Chrome Tabs. 
Turns out there was a change in Android SDK v>23.
See : [Stackoverflow](https://stackoverflow.com/questions/57229179/why-packagemanager-queryintentactivities-method-returning-only-default-browser-i)

The resolveInfoList should be populated like this:
`if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            resolveInfoList = pm.queryIntentActivities(browserIntent,PackageManager.MATCH_ALL );
        }else{
            resolveInfoList = pm.queryIntentActivities(browserIntent, 0);
        }`

I verified in debugger that it solves the problem.

**Comment by FeiChen-okta** [comment]
Hi @maor121, I've tested with fire fox and samsung's browser and both works for me.
Can you try these commands and paste the results:
```bash
adb shell pm query-activities -a android.intent.action.VIEW -d "https://www.example.com"
adb shell pm query-services -a android.support.customtabs.action.CustomTabsService
```
It should list Chrome in both commands. We didn't add MATCH_ALL flag because the documentation states that it should be used with _extreme care_. We will have to understand why package manager is filtering the chrome app you've installed first since it is not filtering the browsers I'm testing with. In the mean time can you try other browsers like firefox and add it to the config like the following:
https://github.com/okta/okta-oidc-android#webauthclient

**Comment by albka1986** [comment]
I have the same issue on Samsung Galaxy J7 (2016)
Config:
```
webClient = Okta.WebAuthBuilder()
            .withConfig(config)
            .withContext(applicationContext)
            .withStorage(SharedPreferenceStorage(applicationContext))
            .create()
```

I tried to add supported browsers, but it didn't help me.

```
webClient = Okta.WebAuthBuilder()
            .withConfig(config)
            .withContext(applicationContext)
            .withStorage(SharedPreferenceStorage(applicationContext))
            .supportedBrowsers("com.android.chrome", "org.mozilla.firefox")
            .create()
```

**Comment by FeiChen-okta** [comment]
Hi @albka1986,

Are you installing the browsers using the play store? The browsers are being filtered due to domain verification failure. The only way to work around this is to use the MATCH_ALL flag. We will have to evaluate the risk first before deciding to add it. You are free to fork this repo and add it yourself.

**Comment by albka1986** [comment]
> Hi @albka1986,
> 
> Are you installing the browsers using the play store? The browsers are being filtered due to domain verification failure. The only way to work around this is to use the MATCH_ALL flag. We will have to evaluate the risk first before deciding to add it. You are free to fork this repo and add it yourself.

On the device, I have default browser from Samsung and Chrome (it was installed already by Samsung)

**Comment by FeiChen-okta** [comment]
Fixed in 1.0.5
Defaults to first available CCT browser if preferred list is empty.

**Comment by maor121** [comment]
@FeiChen-okta Thank you very much for the fix.
I believe there is still a bug though.

The fix defaults to default browser **if default browser supports CCT** .
Details:
`List<ResolveInfo> resolveInfoList = pm.queryIntentActivities(browserIntent, 0);`
Returns only the default browser, and since available customTabsBrowsers is only a subset of resolveInfoList. No browser other then default can be selected. No matter what I do. If I populate preferred list or not.

On my phone (Xiaomi Redmi Note 6 Pro), the default browser doesn't support CCT. So I get error, even though Chrome is installed from PlayStore. 

I had to change my default browser, nothing else worked.
